llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -893,6 +929,16 @@ processInputModule(std::unique_ptr<Module> M) {
 
       ++ID;
     }
+
+    bool dummyEmitted = false;
+    for (module_split::ModuleDesc &IrMD : MMs) {
+      if (auto Dummy = makeDummy(IrMD)) {
+        saveModule(Tables, *Dummy, ID, OutIRFileName, /*IsDummy*/ true);
+        dummyEmitted = true;
+      }
+    }
+    if (dummyEmitted)
+      ++ID;
```

Comment:
We either need some comments here, or the function should be renamed (and likely we need both).

For an unfamiliar reader it could be a mystery what "dummy" means and why do we need to make it for every device image we have produced.

Also: do we have any information about device image carrying virtual function definitions in `ModuleDesc`, or any of its members? If we had an `if` like that at this level, it would save us function arguments lookup and would work as an extra documentation piece